nim/httpbeast/server.nim

```diff
@@ -11,4 +11,4 @@ proc onRequest(req: Request): Future[void] =
     if req.path.get() == "/user":
       req.send(Http200, "")
 
-run(onRequest, Settings(port: Port(3000)))
+run(onRequest, Settings(port: Port(3000), domain: Domain.AF_INET))
```

Comment:
Oh I see. Well here is your bug, please use `initSettings` instead which will set the defaults properly.

Reply:
oh I see ... https://github.com/dom96/httpbeast/blob/b33dba93a134faa2adc3b991cb69890375987076/src/httpbeast.nim#L464 should not be used directly unless using `initSettings`

however, should not `httpbeast` use `AF_UNSPEC` in order to guess if **IPv6** or **IPv4** should be used (depending on system ability) ?

Reply:
@mattaylor should not it be used in ? https://github.com/mattaylor/whip/blob/0.2/whip.nim#L118

Reply:
> however, should not httpbeast use AF_UNSPEC so as either IPv6 or IPv4 is determined on nim level (depending on system ability) ?

not sure what you mean. You mean it should automatically choose IPv6 or IPv4 if `AF_UNSPEC` is used? That's not something Nim supports

Reply:
Exactly @dom96. I think to follow standards **httpbeast** should use `AF_UNSPEC` as long as **nim** should choose IPv4 or IPv6 depending on system abilities https://man7.org/linux/man-pages/man3/getaddrinfo.3.html

Reply:
@waghanza but that's exactly what doesn't work, right? Maybe it's a problem on the Nim side, is that what you think?

Reply:
Indeed. The network / transport layer is, in my opinion, a language concern.